### ISSUE TITLE
General input questions

### ISSUE THREAD

**Issue by manolis07gr** [tardis-sn/tardis#1000]
Hi all again,

In continuation from Issue #999 I also have some additional questions with regards to the TARDIS input in general and I would appreciate some help.

1) I use a MESA Type IIP SN ejecta profile in an effort to get a IIP-looking spectrum using TARDIS. I know the primary development purpose of TARDIS was to investigate Type Ia spectra, but I think making the transition to core-collapse SN ejecta should not be an issue, correct?

2) I am still confused with regards to the reference times. In the input .csvy file (taken from MESA), I am using 

model_density_time_0: 2 d
model_isotope_time_0: 2 d

where the time "2 days" corresponds directly to the time after shock break-out in the MESA model (same phase as in MESA, after shock break out). I think this is right, correct?

Yet I can adjust the time_explosion in my .yml file accordingly to simulate any phase, right? If I set time_explosion = 10 days in my .yml configuration file, the code will homologously expand my input structure out to t = 10 days using the input 2.0 days as reference, correct? That means that if I set time_explosion = 2 days, I should get a model spectrum that corresponds to the same phase as the MESA model, right?

3) I am unsure as to what is the best way to set luminosity_requested. MESA computes the emitted luminosity from the surface at the phase of the model that I am extracting (2 days), so my thinking is that this is the value that I should use. However, at t = 10 days, that luminosity would be different. Is that a free parameter or it can be set based on some physically motivated reasons?

4) I also would like to enable nLTE treatment just for the H, He species. Not sure if that nLTE feature is available yet in TARDIS but, as referenced in issue #999 , I get an error when setting:

#plasma:
    ionization: nebular
    excitation: dilute-lte
    radiative_rates_type: dilute-blackbody
    line_interaction_type: macroatom
    nlte:
        species: ['H I','He II']

In my .yml config file.

In short I am hoping to be able to use TARDIS to model IIP SN spectra (showing the classic P-Cygni Halpha profile, along with other prominent H lines), but I would also like to explore the nLTE option. Any help would be greatly appreciated. I can provide the current input .yml and .csvy files I am using, if necessary.

Thank you,

Manos Chatzopoulos (manolis07gr)

**Comment by manolis07gr** [comment]
@marxwillia  you have been very helpful and I would really appreciate your help with this issue, and with issue #999 .

Thanks,

manolis07gr

**Comment by marxwillia** [comment]
Hi Manos,

Your understanding of the time_explosion, model_density_time_0, model_isotope_time_0 parameters is correct.  The model_density_time_0, model_isotope_time_0 parameters should be set to the time after explosion where your density profile and isotope abundances are defined.

As for using TARDIS to simulate a Type IIP Supernova, you should look at the following papers:
1. https://ui.adsabs.harvard.edu/abs/2019arXiv191104444V/abstract
2. https://ui.adsabs.harvard.edu/abs/2019A%26A...621A..29V/abstract

The extension of TARDIS to model Type IIP requires a more thorough treatment of Hydrogen.  @chvogl has already implemented the physics required, but this is a major update to TARDIS and has not been incorporated into the master branch yet.  

As for the luminosity requested, if you are modelling an observed spectrum, I would use the observations to set the luminosity requested.  Otherwise, it is a free parameter.

I have used the nLTE for Helium in the past without difficulty.  However in order to address your error from #999, I will need to try to reproduce your error.  I will try to get to this soon!

**Comment by manolis07gr** [comment]
@marxwillia thank you for being so incredibly helpful and responsive. If you have any ideas on when the updated TARDIS (to treat H for Type IIP ejecta) will be available, it would be great! Some of the models I plan to run are actually H-poor CCSN ejecta (of Type Ib/Ic), so, I am assuming there won't be significant issues for these, correct? Or am I missing something.

I appreciate you for looking into my #999 error. I can send you the full .csvy and .yml files (unfortunately I can't upload them here - at least in a good format - but I could send them to you in another way you prefer).

Best,

Manos

**Comment by marxwillia** [comment]
It would be helpful if you could upload your files.  Github supports .zip and .gz, so you can just put your files in a temp directory and tar it.  

If your CCSN is Hydrogen poor, then TARDIS should be ok.  Make sure that you use the helium treatment in Boyle et al. (https://www.aanda.org/articles/aa/abs/2017/03/aa29712-16/aa29712-16.html).  We should perhaps talk in more detail either through email or skype since I am also working on modelling CCSNe with TARDIS!  You can contact me at mew488@nyu.edu

**Comment by manolis07gr** [comment]
There you go! Both my .yml and .csvy are attached. The example.py script is the code I use to run the TARDIS simulation and plot the spectrum. It also brings up the GUI (I fixed the bugs that prevented me from doing so before) - but if that gives you trouble you can just delete the lines that enable the GUI.

Does TARDIS have an option to use the helium treatment? If so I would be more than happy to know it (I am relatively new in TARDIS so I am still catching up with all the great things it can do).

Thanks again for the great help, looking forward to your insights,

Manos
[manos_IIp.tar.gz](https://github.com/tardis-sn/tardis/files/3975774/manos_IIp.tar.gz)

**Comment by marxwillia** [comment]
Thanks! In the plasma section of your .yml, set:

plasma:
    helium_treatment: recomb-nlte.  

I recommend looking through the schema files in tardis/io/schemas directory.  This helped me when I was getting started using TARDIS to understand what options are available.

**Comment by manolis07gr** [comment]
Hi @marxwillia, I was wondering if you had the chance to run my yml/csvy files through and see yourself what I miss. While this SN ejecta has Hydrogen, which we now know is important since TARDIS (public version) is still missing the relevant physics, it may be a good starting point for testing. My next step is to run a Ib/Ic SN ejecta simulation with the aim to get these spectra right. Any feedback we can exchange on this type of modeling would be useful.
